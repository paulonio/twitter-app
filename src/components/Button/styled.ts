import { css, styled } from 'styled-components';

import type { ButtonType } from './Button';

interface StyledButtonProps {
  $buttonType: ButtonType;
}

export const StyledButton = styled.button<StyledButtonProps>`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 50px;
  font-size: 18px;
  line-height: 100%;
  font-weight: 700;
  transition: background-color 0.3s;
  &:hover {
    cursor: pointer;
  }
  ${(props) => {
    switch (props.$buttonType) {
      case 'primary':
        return css`
          padding: 15px 0;
          background-color: #1d9bf0;
          color: #ffffff;
          &:hover {
            background-color: #4fa9e6;
          }
        `;
      case 'secondary':
        return css`
          padding: 10px 0;
          background-color: ${({ theme }) => theme.colors.backgroundSecondaryButton};
          color: ${({ theme }) => theme.colors.textSecondaryButton};
          &:hover {
            background-color: ${({ theme }) => theme.colors.backgroundSecondaryButtonHover};
          }
        `;
      case 'bordered':
        return css`
          padding: 10px 15px;
          color: ${({ theme }) => theme.colors.text};
          background-color: ${({ theme }) => theme.colors.backgroundBody};
          border: solid 1px ${({ theme }) => theme.colors.borderButton};
          line-height: 24px;
          &:hover {
            background-color: ${({ theme }) => theme.colors.backgroundborderedButtonHover};
            cursor: pointer;
          }
        `;
      case 'link':
        return css`
          width: max-content;
          font-size: 18px;
          line-height: 100%;
          font-weight: 400;
          color: #1da1f2;
          background-color: transparent;
          border: none;
          transition: color 0.3s;
          &:hover {
            cursor: pointer;
            color: #4fa9e6;
          }
        `;
      default:
        return css``;
    }
  }}
`;
