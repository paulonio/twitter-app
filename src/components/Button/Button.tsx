import React, { FC, HTMLAttributes } from 'react';

import { StyledButton } from './styled';

export type ButtonType = 'primary' | 'secondary' | 'bordered' | 'link' | 'danger';

type ButtonProps = {
  children?: string;
  $buttonType: ButtonType;
} & HTMLAttributes<HTMLButtonElement>;

const Button: FC<ButtonProps> = (props) => {
  return <StyledButton {...props} />;
};

export default Button;
