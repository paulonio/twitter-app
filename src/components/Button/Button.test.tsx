import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Button from '@components/Button/Button';

describe('Button component', () => {
  it('should render a button', () => {
    const { container } = render(<Button $buttonType="primary" />);

    expect(container).toMatchSnapshot();
  });
});
