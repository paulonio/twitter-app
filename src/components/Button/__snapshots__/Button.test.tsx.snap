// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render a button 1`] = `
<div>
  <button
    class="sc-erFrCC iFaZZO"
  />
</div>
`;
